test/recase_test.dart

```diff
@@ -64,6 +88,11 @@ void main() {
     test('from "${allCapsInput.originalText}".', () {
       expect(allCapsInput.headerCase, equals('Foo-Bar'));
     });
+
+    test('from "${acronynInput.originalText}".', () {
+      expect(
+          acronynInput.headerCase, equals('My-Http-DB-Connection-Info-Id-Xml'));
```

Comment:
I'm not entirely sure if header case (or title case etc.) should follow the same rules of capitalization as those laid out for camel case by the "Effective Dart" style guide.  The current behavior is to do so but I'm not married to that.